### PR TITLE
Logout logic + component

### PR DESCRIPTION
**Pull request by BlueEyedHush**

### CODE REVIEW

File: src/node_modules/client/app/components/navbar/IconNavItem.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import { NavItem, Glyphicon } from 'react-bootstrap'
+
+export default class IconNavItem extends React.Component {
```
Comment (quirk0o) [inline src/node_modules/client/app/components/navbar/IconNavItem.jsx]: Why change the name?
What was wrong with the previous one?

File: src/node_modules/client/app/components/navbar/IconLink.jsx [hunk 2]
```diff
@@ -1,32 +0,0 @@
...
-  constructor(props) {
-    super(props);
-  }
-
-  render() {
-    let { to, glyph, text, ...props } = this.props;
-    let desc = text ? <span className="navbar-icon-desc">{text}</span> : '';
-    let style = this.props.className ? `${this.props.className} navbar-icon` : 'navbar-icon';
-    return (
-        <LinkContainer to={to}>
```
Comment (quirk0o) [inline src/node_modules/client/app/components/navbar/IconLink.jsx]: What's the reason for this change?
What was wrong with this implementation?

File: src/node_modules/client/app/components/navbar/Navbar.jsx [hunk 3]
```diff
@@ -72,30 +73,37 @@ export default class MainNavbar extends React.Component {
...
+              <LinkContainer to="/settings">
+                <IconNavItem
+                  glyph="cog"
+                  text="Settings"
+                  eventKey={7}
+                  onClick={this.collapseNav}
+                />
+              </LinkContainer>
+
+              <Logout eventKey={8} />
```
Comment (quirk0o) [inline src/node_modules/client/app/components/navbar/Navbar.jsx]: Why is logout link both in Settings and here?
It should only be in Settings imho

Comment (BlueEyedHush) [inline src/node_modules/client/app/components/navbar/Navbar.jsx]: I wanted it to appear in Settings Dropdown here:
image
You think it should be outside of the dropdown?

File: .eslintrc [hunk 4]
```diff
@@ -38,6 +38,7 @@
     "func-names": "off",
     "arrow-parens": "off",
 
+    "import/imports-first": "off",
```
Comment (quirk0o) [inline .eslintrc]: Why?

File: src/node_modules/client/app/components/navbar/IconNavItem.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import {NavItem, Glyphicon} from 'react-bootstrap';
+
+const IconNavItem = (props) => {
+  const {glyph, text, className, ...restOfProps} = props;
+  const desc = text ? <span className="navbar-icon-desc">{text}</span> : '';
+  const style = className ? `${className} navbar-icon` : 'navbar-icon';
+  return (
+    <NavItem {...restOfProps} className={style}>
```
Comment (quirk0o) [inline src/node_modules/client/app/components/navbar/IconNavItem.jsx]: Sooo you changed my `props` into `restOfProps`? Why is that better. If you want to change it do something descriptive like `navItemProps`

File: src/node_modules/client/app/components/navbar/Logout.jsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+import IconNavItem from './IconNavItem';
+
+import {logout} from 'client/sharedjs/auth';
+
+const Logout = (props) => {
```
Comment (quirk0o) [inline src/node_modules/client/app/components/navbar/Logout.jsx]: `LogoutNavItem`?

Comment (quirk0o) [inline src/node_modules/client/app/components/navbar/Logout.jsx]: And why does it need a separate component if none of the other nav items have one?
